tests/test_integration_real_llm.py

```
@@ -0,0 +1,127 @@
+    async def llm_setup(self):
+        """Setup LLM providers for integration tests."""
+        google_api_key = os.getenv("GOOGLE_API_KEY")
+        if not google_api_key:
+            pytest.skip("GOOGLE_API_KEY not set - skipping real LLM integration tests")
+        
+        await setup_llm_providers(google_api_key=google_api_key)
+
+    @pytest.mark.asyncio
+    @pytest.mark.skipif(
+        not os.getenv("GOOGLE_API_KEY"),
+        reason="GOOGLE_API_KEY not set - requires real API access"
```

Ruff (0.12.2)

Trailing comma missing

Add trailing comma

(COM812)